hr_expense_petty_cash/models/hr_expense.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class HrExpense(models.Model):
+    _inherit = 'hr.expense'
+
+    payment_mode = fields.Selection(
+        selection_add=[('petty_cash', 'Petty Cash')],
+    )
+    petty_cash_id = fields.Many2one(
+        string='Petty cash holder',
+        comodel_name='petty.cash',
+        ondelete='restrict',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+
+    @api.multi
+    def action_submit_expenses(self):
+        res = super().action_submit_expenses()
+        context = res.get('context', False)
```

Comment:
Here you try to get the `context` and default to `False` if the context does not exist in `res` but below you do not take into consideration the fact that context can be `False`, either do that or go ahead and `res['context']`